numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1169,7 +1187,23 @@ arr_unravel_index(PyObject *self, PyObject *args, PyObject *kwds)
     unravel_size = PyArray_MultiplyList(dimensions.ptr, dimensions.len);
 
     if (!PyArray_Check(indices0)) {
-        indices = (PyArrayObject*)PyArray_FROM_O(indices0);
+        /* prefer int dtype */
+        PyArray_Descr *dtype_guess = NULL;
+        if (PyArray_DTypeFromObject(indices0, NPY_MAXDIMS, &dtype_guess) < 0) {
+            goto fail;
+        }
+        if (dtype_guess == NULL) {
+            if (PySequence_Check(indices0) && PySequence_Size(indices0) == 0) {
+                PyErr_SetString(PyExc_TypeError,
+                                "empty sequence implies float indices "
+                                "which are not allowed");
+            }
+            else if (!PyErr_Occurred()) {
+                PyErr_SetString(PyExc_RuntimeError, "unknown error occurred");
+            }
```

Comment:
This seems like it should never happen, and we should leave python to raise the usual `SystemError` here if it does

Reply:
codecov agrees with you, those lines were never hit. Removing.